sympy/series/order.py

```diff
@@ -415,6 +416,10 @@ def _eval_subs(self, old, new):
                         from sympy.solvers.solveset import solveset
                         d = Dummy()
                         sol = solveset(old - new.subs(var, d), d)
+                        if isinstance(sol, Complement):
```

Comment:
@hargup Sorry, there is a confusion between us. Those tests were already `not` having `@XFAIL`. My first commit failed some of the tests, related to the `series module` raising an error of `zip a non-interable`. My second commit fixes that problem present in my first commit. You can look those failing `tests` in the first commit, `travis-build`.